### ISSUE TITLE
Modal inside Modal: App got frozen when dismiss parent Modal

### ISSUE THREAD

**Issue by olegmilan** [facebook/react-native#28343]
**Description:**

As I [mentioned](https://stackoverflow.com/q/60752730/8165679) in stackoverflow: there's no proper way in React-Native to show Modal on top of the Modal and I use [workaround](https://stackoverflow.com/a/35055128/8165679) to achieve this.

But there's another issue: when I **dismiss parent modal** app got frozen(on **IOS**, didn't check Android, but I need both platforms to be working) and parent modal remain opened with frozen UI and you have to force close the app to unblock UI

**React Native version:**

Currently it's `0.60.4`, but I tried with different versions and this issue appears for a long time

## Steps To Reproduce
   
1. Have a Modal inside Modal
2. Dismiss parent(main) Modal

## Expected Results

Modals dismiss properly without adding hacks like dismissing Child modal and then Parent modal
## Code example and GIF screenshot:

Example:
![modal-inside-modal-issue](https://user-images.githubusercontent.com/31205092/77043965-3f527b80-69c7-11ea-8c2b-6831a658eee7.gif)

Code:

 ```
import React, { Component } from 'react';
import { Modal, View, TouchableHighlight, Text } from 'react-native';

class MultipleModal extends Component {
  state = {
    parentModalVisible: false,
    childModalVisible: false,
  };

  closeOpenModal = (parentModalVisible) => {
    this.setState({
      parentModalVisible,
      childModalVisible: parentModalVisible,
    });
  };

  renderModals() {
    const { parentModalVisible, childModalVisible } = this.state;
    return (
      <Modal
        visible={parentModalVisible}
        transparent={false}
        animationType={'none'}
        onRequestClose={() => this.closeOpenModal(false)}
      >
        <View style={{ flex: 1, backgroundColor: 'red' }} />
        <Modal
          visible={childModalVisible}
          transparent={false}
          animationType={'none'}
          onRequestClose={() => this.closeOpenModal(false)}
        >
          <View style={{ flex: 1, backgroundColor: 'green' }}>
            <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
              <TouchableHighlight
                style={{ width: 100, height: 40, justifyContent: 'center', alignItems: 'center', backgroundColor: 'lightblue' }}
                onPress={() => this.closeOpenModal(false)}
              >
                <Text>Close Modal</Text>
              </TouchableHighlight>
            </View>
          </View>
        </Modal>
      </Modal>
    );
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
          <TouchableHighlight
            style={{ width: 100, height: 40, justifyContent: 'center', alignItems: 'center', backgroundColor: 'lightblue' }}
            onPress={() => this.closeOpenModal(true)}
          >
            <Text>Open Modal</Text>
          </TouchableHighlight>
        </View>
        {this.renderModals()}
      </View>
    );
  }
}
```

Currently to avoid this issue I have to close modals **asynchronously**, first child and then parent which is a bit painful and adds complexity to the code and possible issues:

```
  closeModal = () => {
    // Dismiss child
    this.setState({ childModalVisible: false }, () => {
      // Dismiss parent, but add requestAnimationFrame to prevent app from being frozen
      requestAnimationFrame(() => {
        this.setState({ parentModalVisible: false });
      });
    });
  };
```

**Comment by Foveluy** [comment]
the problem comes from here [https://github.com/facebook/react-native/pull/27618](url)，check out this PR. Quick way to solve this problem is that copy the code from this pr. @olegmilan
